src/dos/cdrom_image.cpp

```diff
@@ -530,26 +852,16 @@ bool CDROM_Interface_Image::LoadCueSheet(char *cuefile)
 			if (type == "BINARY") {
 				track.file = new BinaryFile(filename.c_str(), error);
 			}
-#if defined(C_SDL_SOUND)
-			//The next if has been surpassed by the else, but leaving it in as not 
-			//to break existing cue sheets that depend on this.(mine with OGG tracks specifying MP3 as type)
-			else if (type == "WAVE" || type == "AIFF" || type == "MP3") {
-				track.file = new AudioFile(filename.c_str(), error);
-			} else { 
-				const Sound_DecoderInfo **i;
-				for (i = Sound_AvailableDecoders(); *i != NULL; i++) {
-					if (*(*i)->extensions == type) {
-						track.file = new AudioFile(filename.c_str(), error);
-						break;
-					}
-				}
-			}
-#endif
-			if (error) {
-				delete track.file;
-				track.file = NULL;
-				success = false;
-			}
+            else
+                track.file = new AudioFile(filename.c_str(), error);
```

Comment:
I think you noticed already, but Clang static analyzer indicates a memory leak in here (the interaction triggering memory leak is very complicated and requires at least 3 loop iterations, IIRC. Anyways, I think it can be easily resolved by using C++11 `std::unique_ptr`.

Reply:
If `track.file` was created in the local-scope, then this would be a textbook use, but `track.file` is a member variable (and a common one for two derived classes), so we'd have to change its type to `unique_ptr` up in the class definition.  I'm a bit reluctant only because I'm worried it will be another "mark against" the code for merging upstream.. but depending on what acceptance we see, and if upstream is willing to accept these C++ "upgrades", then I will go through this Audio code and move all of the member pointers to smart pointers like this (and do-away with delete across the board).  We'll see!

Reply:
"won't fix" for now, but tentative for future changes that "upgrade" to newer C++'isms.

Reply:
PS- these two "fromISO" functions are a mess of spaghetti.. looking forward to re-writing them too.